controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -58,62 +74,180 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
-	readyCondition, reportedError := r.determineReadyState(ctx, spaceProvisionerConfig)
+	reportedErr := r.refreshStatus(ctx, spaceProvisionerConfig)
 
-	var updated bool
-	spaceProvisionerConfig.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(spaceProvisionerConfig.Status.Conditions,
-		readyCondition)
-	if !updated {
-		return ctrl.Result{}, reportedError
+	if err := r.Client.Status().Update(ctx, spaceProvisionerConfig); err != nil {
+		return ctrl.Result{}, err
 	}
 
-	logger.Info("updating SpaceProvisionerConfig", "readyCondition", readyCondition)
-	if err := r.Client.Status().Update(ctx, spaceProvisionerConfig); err != nil {
-		if reportedError != nil {
-			logger.Info("failed to update the status (reported as failed reconciliation) with a previous unreported error during reconciliation", "unreportedError", reportedError)
+	return ctrl.Result{}, reportedErr
+}
+
+func (r *Reconciler) refreshStatus(ctx context.Context, spc *toolchainv1alpha1.SpaceProvisionerConfig) error {
+	// clear out the consumed capacity - this will advertise to the user that we either failed before it made sense
+	// to collect it (and therefore we don't know it) or it was not available (and therefore we again don't know it)
+	spc.Status.ConsumedCapacity = nil
+
+	if !spc.Spec.Enabled {
+		updateReadyCondition(spc, corev1.ConditionFalse, toolchainv1alpha1.SpaceProvisionerConfigDisabledReason, "")
+		return nil
+	}
+
+	clusterCondition, err := r.determineClusterReadyState(ctx, spc)
+	if err != nil {
+		updateReadyCondition(spc, clusterCondition, toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason, err.Error())
+
+		// the reconciler reacts on ToolchainCluster changes so it will be triggered once a new TC appears
+		// we therefore don't need to return error from the reconciler in the case the TC is not found.
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	var reason string
+	resultCondition := clusterCondition
+
+	if clusterCondition == corev1.ConditionTrue {
+		collectUsage := r.GetUsageFunc
+		if collectUsage == nil {
+			collectUsage = collectConsumedCapacity
+		}
+		cc, err := collectUsage(ctx, r.Client, spc.Spec.ToolchainCluster, spc.Namespace)
+		if err != nil {
+			updateReadyCondition(spc, corev1.ConditionUnknown, toolchainv1alpha1.SpaceProvisionerConfigFailedToDetermineCapacityReason, err.Error())
+			return err
+		}
+		spc.Status.ConsumedCapacity = cc
+
+		capacityCondition := r.determineCapacityReadyState(spc)
```

Comment:
I considered that but since both the limits and the usage are in the same CR, it's a matter of just glancing at the data to see what's wrong. IMHO, if we wanted to be this detailed, we should add separate conditions for different aspects of the readiness computation.

So far, a single condition was enough for us, so I didn't want to overdo the error reporting when all the information is already in the CR.